kubernetes/resource_kubectl_manifest.go

```
@@ -829,26 +829,32 @@ func GetFingerprint(s string) string {
 	if userProvided.GetKind() == "Secret" && userProvided.GetAPIVersion() == "v1" {
-		if stringData, found := userProvided.Raw.Object["stringData"]; found {
+		if stringData, found := userObject["stringData"]; found {
 			// there is an edge case where stringData might be nil and not a map[string]interface{}
 			// in this case we will just ignore it
 			if stringData, ok := stringData.(map[string]interface{}); ok {
-				// move all stringdata values to the data
+				userObject = userProvided.Raw.DeepCopy().Object
 				for k, v := range stringData {
 					encodedString := base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf("%v", v)))
-					meta_v1_unstruct.SetNestedField(userProvided.Raw.Object, encodedString, "data", k)
+					meta_v1_unstruct.SetNestedField(userObject, encodedString, "data", k)
```

golangci-lint (2.12.2)

[error] Error return value of `meta_v1_unstruct.SetNestedField` is not checked

(errcheck)